### PR TITLE
Yarn scripts for quality of life in development

### PR DESCRIPTION
**Pull request by Iku-turso**
Run and watch targetable unit- or integration-tests.

### CODE REVIEW

**Review by Nokel81 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -28,6 +28,8 @@
     "build:mac": "yarn run compile && electron-builder --mac --dir",
     "build:win": "yarn run compile && electron-builder --win --dir",
     "integration": "jest --runInBand --detectOpenHandles --forceExit integration",
+    "test:unit": "jest --watch --testPathIgnorePatterns integration",
```
Comment (jim-docker) [inline package.json:31]: I discovered you need to make sure you have an up-to-date build before running this

File: package.json [hunk 2]
```diff
@@ -28,6 +28,8 @@
...
+    "test:integration": "func() { jest ${1:-xyz} --watch --runInBand --detectOpenHandles --forceExit --modulePaths=[\"<rootDir>/integration/\"]; }; func",
```
Comment (jim-docker) [inline package.json:32]: What file changes will cause tests to rerun? I edited/saved some files but no tests were run ("No tests found, exiting with code 0").

File: package.json (hunk 1 above)
Comment (Iku-turso) [reply package.json:31]: Yes, this is intentional. The script is for development, and for that, it is optimized for quick start-up.

File: package.json (hunk 2 above)
Comment (Iku-turso) [reply package.json:32]: For `unit tests`, watch triggers after a change in most production or test code (and also some configuration files). The initial behavior of `jest --watch` is to use git to find uncommitted production and test-code, and run all tests directly or indirectly affected by those changes. 

In your case: 
1. The changed files need to be related to a test for anything to happen.
2. Our current unrelated configuration expects `*.test`-files to contain at least one test in order to not be perceived as failure. This explains `"No tests found, exiting with code 0"`.

Deviating from this, for `integration tests` the watch triggers only for changes in test code, as the tests happen over already built application. Still, this makes tweaking the slow integration tests a bit more friendly :)

FYI: In watch mode, you can also use `w` to see additional controls. Eg. `p` or `t` will make you able to specify explicitly which tests to run, `a` will make you run all tests despite git, and `f` will make you re-run only failing tests. Cool stuff, right? :)

File: package.json (hunk 1 above)
Comment (jim-docker) [reply package.json:31]: ok, bit messy if it's already running when you checkout a new branch

File: package.json (hunk 2 above)
Comment (jim-docker) [reply package.json:32]: pretty sure I edited an integration test file (edited some string, maybe the test name). I'll try again

Comment (jim-docker) [reply package.json:32]: I added a test to `integration/__tests__/cluster-pages.tests.ts` and saved. No tests found. Pressing `o` runs the tests. Is this how to use it?

**Review by jim-docker — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (Iku-turso) [reply package.json:32]: We could have a sesh? Or all this should be documented somewhere within jest. Hmm.

**Pull request merged**